src/WebAuthn.sol

```diff
@@ -1,32 +1,29 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.0;
 
-import {Base64Url} from "FreshCryptoLib/utils/Base64Url.sol";
+import {Base64} from "solady/utils/Base64.sol";
 import {FCL_ecdsa} from "FreshCryptoLib/FCL_ecdsa.sol";
+import {LibString} from "solady/utils/LibString.sol";
 
 /// @title WebAuthn
 /// @notice A library for verifying WebAuthn Authentication Assertions, built off the work
-/// of Daimo. This library is optimized for calldata,
-/// and attempts to use the RIP-7212 precompile for signature verification.
+/// of Daimo.
+/// @dev Attempts to use the RIP-7212 precompile for signature verification.
 /// If precompile verification fails, it falls back to FreshCryptoLib.
 /// @author Coinbase (https://github.com/base-org/webauthn-sol)
 /// @author Daimo (https://github.com/daimo-eth/p256-verifier/blob/master/src/WebAuthn.sol)
 library WebAuthn {
+    using LibString for string;
+
     struct WebAuthnAuth {
         /// @dev https://www.w3.org/TR/webauthn-2/#dom-authenticatorassertionresponse-authenticatordata
         bytes authenticatorData;
-        /// @dev https://www.w3.org/TR/webauthn-2/#dom-collectedclientdata-origin
-        string origin;
-        /// @dev https://www.w3.org/TR/webauthn-2/#dom-collectedclientdata-crossorigin
-        /// @dev 13. https://www.w3.org/TR/webauthn/#clientdatajson-serialization
-        /// crossOrigin should always be present, re https://www.w3.org/TR/webauthn/#clientdatajson-serialization
-        /// but in practice is sometimes not. For this reason we include with remainder. String may be empty.
-        /// e.g.
-        ///     ''
-        ///     '"crossOrigin":false'
-        ///     '"tokenBinding":{"status":"present","id":"TbId"}'
-        ///     '"crossOrigin":false,"tokenBinding":{"status":"present","id":"TbId"}'
-        string crossOriginAndRemainder;
+        /// @dev https://www.w3.org/TR/webauthn-2/#dom-authenticatorresponse-clientdatajson
+        string clientDataJSON;
+        /// The index at which "challenge":"..." occurs in clientDataJSON
+        uint256 challengeIndex;
+        /// The index at which "type":"..." occurs in clientDataJSON
+        uint256 typeIndex;
```

Comment:
I have one thing I've been thinking about recently. Technically, compared to the previous implementation, there's less of a guarantee that the `"challenge":"..."` and `"type":"..."` fields are actually occurring in the top level of the JSON structure. For example if a future version of WebAuthn has a field called `arbitraryData` in the `clientDataJSON`, then I'm imagining a JSON like this:

```javascript
{
    "type":"webauthn.get",
    "challenge": "<challenge1>",
    "arbitraryData": "\"challenge\": \"<challenge2>\""
}
```

With this example, the arbitrary data contains a string that looks like a `challenge` field in the JSON, but really it's just a random string. In this scenario it might be possible to point the `challengeIndex` to the arbitrary string to prove a challenge the user wasn't expecting.

I think the implication here is that users need to be careful of everything they're signing, and they need to avoid `clientDataJSON` that have obvious red flags like the above. Maybe adding a note about this in the comments would be useful.

WDYT? The rest of the PR looks good to me pending this last note